schema_based_on_diagram.sql

```diff
@@ -0,0 +1,56 @@
+-- create database from clinic diagram -- clinic
+CREATE DATABASE clinic;
+
+-- create a patients table
+CREATE TABLE patients (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    name VARCHAR(50),
+    date_of_birth DATE,
+    PRIMARY KEY(id)
+);
+
+-- create medical_histories table
+CREATE TABLE medical_histories (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    admitted_at TIMESTAMP,
+    status VARCHAR(50),
+    CONSTRAINT fk_patients FOREIGN KEY(patient_id) REFERENCES patients(id),
+    PRIMARY KEY (id)
+);
+
+-- create invoices table
+CREATE TABLE invoices (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    total_amount DECIMAL,
+    generated_at TIMESTAMP,
+    payed_at TIMESTAMP,
+    CONSTRAINT fk_medical_history FOREIGN KEY(medical_history_id) REFERENCES medical_histories (id), 
+    PRIMARY KEY (id)
+);
```

Comment:
- [ ] An error is yielding in attempt to create `invoice` table referencing patients `medical_histories`. You can modify _line 27_ by simply creating a foreign key and reference like this, which works:
```sql
 medical_history_id integer REFERENCES medical_histories (id), 
```

**Create invoice table error:**
![image](https://user-images.githubusercontent.com/12745474/186177188-3e9a5648-a50f-4036-8caf-003236a6abb2.png)